test/datadog/lambda/metrics.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/BlockLength
+require 'datadog/lambda'
+require 'datadog/statsd'
+require 'datadog/lambda/metrics'
+require 'datadog/lambda/utils/logger'
+
+describe Datadog::Metrics::Client do
+  subject(:client) { Datadog::Metrics::Client.instance }
+  let(:layer_tag) { Datadog::Lambda.dd_lambda_layer_tag }
+  let(:default_tags) { ["dd_lambda_layer:datadog-ruby#{layer_tag}"] }
+
+  describe '.instance' do
+    it 'returns the same instance' do
+      instance1 = Datadog::Metrics::Client.instance
+      instance2 = Datadog::Metrics::Client.instance
+      expect(instance1).to be(instance2)
+    end
+  end
+
+  describe '#distribution' do
+    context 'when extension is running' do
+      let(:statsd) { instance_double(Datadog::Statsd) }
+
+      before(:each) do
+        # Mock the extension_running? method to return true
```

Comment:
nit: but is this more of a stub? Maybe my understandings are off. Stubs and mocks seem really similar

Reply:
I'm stubbing a method yeah, I just called it "mocking", I'll change the comments to be more descriptive, good catch!